crates/proc_macro_srv/src/rustc_server.rs

```diff
@@ -185,38 +193,38 @@ pub mod token_stream {
                 mbe::parse_to_token_tree(src).ok_or("Failed to parse from mbe")?;
 
             let subtree = subtree_replace_token_ids_with_unspecified(subtree);
-            Ok(TokenStream { subtree })
+            Ok(TokenStream::with_subtree(subtree))
         }
     }
 
     impl ToString for TokenStream {
         fn to_string(&self) -> String {
-            let tt = self.subtree.clone().into();
-            to_text(&tt)
+            tokentrees_to_text(&self.token_trees[..])
         }
     }
 
-    fn to_text(tkn: &tt::TokenTree) -> String {
+    fn tokentrees_to_text(tkns: &[tt::TokenTree]) -> String {
```

Comment:
Nit: we usuallyy put inner function at the end of a function.

Reply:
Is https://github.com/rust-analyzer/rust-analyzer/pull/7892/commits/aea974939064b0f7b83de371a93ee4190c80e544 what you meant by this?